controller_manager:
  ros__parameters:
    update_rate: 50

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    joint_group_position_controller:
      type: position_controllers/JointGroupPositionController

    joint_group_effort_controller:
      type: effort_controllers/JointGroupEffortController


joint_state_broadcaster:
  ros__parameters: {}

joint_group_position_controller:
  ros__parameters:
    joints:
      - link1_to_base
      - link2_to_link1
      - link3_to_link2
    command_interfaces:
      - position
    state_interfaces:
      - position
    command_topic: /joint_group_position_controller/command

joint_group_effort_controller:
  ros__parameters:
    joints:
      - link1_to_base
      - link2_to_link1
      - link3_to_link2
    command_interfaces:
      - effort
    state_interfaces:
      - position
      - velocity
    command_topic: /joint_group_effort_controller/command